cluster-state-service/handler/event/kinesis_consumer.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package event
+
+import (
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kinesis/kinesisiface"
+	"time"
+	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+const (
+	kinesisWaitTimeSeconds = 10
+)
+
+type kinesisEventConsumer struct {
+	kinesis kinesisiface.KinesisAPI
+	streamName string
+	processor Processor
+	iterator *string
+}
+
+func NewKinesisConsumer(kinesis kinesisiface.KinesisAPI, processor Processor, streamName string) (Consumer, error) {
+	if kinesis == nil {
+		return nil, errors.Errorf("The Kinesis API interface is not initialized")
+	}
+	if processor == nil {
+		return nil, errors.Errorf("The event processor is not initialized")
+	}
+	if streamName == "" {
+		return nil, errors.Errorf("The Kinesis stream name is empty")
+	}
+
+	return &kinesisEventConsumer{
+		kinesis: kinesis,
+		streamName: streamName,
+		processor: processor,
+	}, nil
+}
+
+func (kinesisConsumer *kinesisEventConsumer) PollForEvents(ctx context.Context) {
+	log.Infof("Starting to poll for events from Kinesis")
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			kinesisConsumer.pollForMessages()
+		}
+	}
+}
+
+func (kinesisConsumer *kinesisEventConsumer) pollForMessages() {
+	log.Infof("In pollForMessages")
+	if kinesisConsumer.iterator == nil {
+		log.Infof("First pass, getting iterator")
+		iteratorRequest := &kinesis.GetShardIteratorInput{
+			ShardId: aws.String("shardId-000000000000"),
```

Comment:
It seems that shardId-000000000000 is some sort of magic value that you are supposed to start with to go to the beginning of a stream.  I'll turn this into a const for now since it's unlikely someone would be passing an alternate value since shard IDs are only guaranteed valid for something like 300 seconds.